package_modules/yum

```diff
@@ -0,0 +1,190 @@
+#!/usr/bin/python
+
+import sys
+import os
+import subprocess
+import re
+
+
+rpm_cmd = os.environ.get('CFENGINE_TEST_RPM_CMD', "/bin/rpm")
+rpm_output_format = "Name=%{name}\nVersion=%{version}\nArchitecture=%{arch}\n"
+
+yum_cmd = os.environ.get('CFENGINE_TEST_YUM_CMD', "/usr/bin/yum")
+yum_options = "--quiet"
+
+def get_package_data():
+    pkg_string = ""
+    for line in sys.stdin:
+        if line.startswith("File="):
+            pkg_string = line.split("=", 1)[1].rstrip()
+            break
+
+    if not pkg_string:
+        return 1
```

Comment:
Note that all your other top-level commands carefully consume all of sys.stdin.
If that isn't necessary, kill their calls to .readlines().
If it necessary, then it is here, too; and you can replace the parentheses for files with square brackets, then use files[0] and catch IndexError instead of using the generator I wrote above.  You'll then be sure to consume all of sys.stdin and you'll quietly discard any excess 'File=' lines.  You could even check for those and grumble about them, if len(files) > 1.